### PR TITLE
Added automatic dash gui generator using dash-component-editor

### PR DESCRIPTION
**Pull request by zhuowenzhao**
Added model selection dropdown and fully integrated dash-component-editor into the interface.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -31,8 +34,26 @@
         control_bar_layout(),
         image_viewer_layout(),
         dcc.Store(id="current-class-selection", data="#FFA200"),
+        dcc.Store(id="gui-components-values", data={})
     ],
 )
 
+### automatic Dash gui callback ###
```
Comment (dylanmcreynolds) [inline app.py:41]: Why is this in app and not in something in callbacks?

Comment (dylanmcreynolds) [reply app.py:41]: Actually, why is this a callback and not a function?

**Review by dylanmcreynolds — COMMENTED** [review]
Left a couple of questions

**Comment by zhuowenzhao** [comment]
I am confused about the " test-and-lint" failures. Not sure what is wrong with the way importing libs.

**Comment by Wiebke** [comment]
The latest updates simplify some of the structure for generating components, use subclassing to the already existing control component already in the application, and remove some code that was either not used or could be called in a simpler way elsewhere.

Two outstanding smaller issues are:
- We are not retaining common settings when switching between models
- There is currently no space between components (fixable through either adding Space components or CSS style adaptions)

**Review by taxe10 — APPROVED** [review]

**Pull request merged**